src/holon/models/scenario_rule.py

```diff
@@ -157,6 +159,24 @@ def get_filtered_repository(self, scenario_aggregate: ScenarioAggregate) -> Repo
         # MAKE SURE RESULTS ARE DISTINCT
         return repository
 
+    def apply_filters_to_repository(
+        self, scenario_aggregate: ScenarioAggregate, repository: RepositoryBaseClass
+    ) -> RepositoryBaseClass:
+        """Apply filters to a repository"""
+
+        # filter the repository for model subtype
+        if self.model_subtype:
+            model_subtype = apps.get_model("holon", self.model_subtype)
+            repository = repository.filter_model_subtype(model_subtype)
+
+        # apply rule filters to repository
+        filters = self.get_filters()
+        for filter in filters:
+            repository = filter.filter_repository(scenario_aggregate, repository)
+
+        # MAKE SURE RESULTS ARE DISTINCT
+        return repository
+
```

Comment:
Deze code lijkt identiek aan het onderste stuk van `get_filtered_repository`. Kunnen we gelijk deze functie vanuit `get_filtered_repository` aanroepen om dubbele code te voorkomen?

Reply:
Deze code zal per ongeluk gedupliceerd zijn. Verwijderd.